contracts/registry.go

```diff
@@ -32,23 +32,33 @@ const (
                              }]`
 )
 
-var getAddressForFuncABI, _ = abi.JSON(strings.NewReader(getAddressForABI))
+var getAddressMethod *BoundMethod
+
+func init() {
+	var err error
+	getAddressForFuncABI, err := abi.JSON(strings.NewReader(getAddressForABI))
+	if err != nil {
+		panic("can't parse registry abi " + err.Error())
+	}
+
+	getAddressMethod = NewBoundMethod(params.RegistrySmartContractAddress, &getAddressForFuncABI, "getAddressFor", params.MaxGasForGetAddressFor)
+}
 
 // TODO(kevjue) - Re-Enable caching of the retrieved registered address
 // See this commit for the removed code for caching:  https://github.com/celo-org/geth/commit/43a275273c480d307a3d2b3c55ca3b3ee31ec7dd.
 
 // GetRegisteredAddress returns the address on the registry for a given id
-func GetRegisteredAddress(evm *vm.EVM, registryId common.Hash) (common.Address, error) {
-	evm.DontMeterGas = true
-	defer func() { evm.DontMeterGas = false }()
+func GetRegisteredAddress(caller vm.SystemEVM, registryId common.Hash) (common.Address, error) {
+	caller.StopGasMetering()
+	defer caller.StartGasMetering()
 
 	// TODO(mcortesi) remove registrypoxy deployed at genesis
```

Comment:
While we're here, remove or clarify this TODO.  And maybe also the ones below (lines 63-64 on this branch.